storage/store.go

```diff
@@ -1382,14 +1387,25 @@ func (s *Store) resolveWriteIntentError(ctx context.Context, wiErr *proto.WriteI
 			},
 		}
 		resolveReply := &proto.InternalResolveIntentResponse{}
-		// Add resolve command with wait=false to add to Raft but not wait for completion.
-		waitForResolve := wait && i == len(wiErr.Intents)-1
-		if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, waitForResolve); resolveErr != nil {
-			if log.V(1) {
-				log.Warningc(ctx, "resolve for key %s failed: %s", intent.Key, resolveErr)
+
+		go func(errCh chan<- error, key proto.Key) {
```

Comment:
We do not want to return resolve errors to the caller; the caller is not set up to handle them properly, nor should it be.

If nothing else has changed at the intent after pushing the txn, then the resolve should work. If the resolve doesn't work, it means something else has happened to the key with the intent. The intent is no longer there (already cleaned up), a new intent is there, key isn't there anymore, etc. None of these mean that this command shouldn't simply be retried again in order to encounter the next error.

By returning resolve errors, you're going to cause the retry loop in Store.ExecuteCmd to exit with the resolve error, prematurely causing the client op / txn to fail on what is a retryable error.